### PR TITLE
tools: update eksctl and eksctl anywhere to enable K8s 1.24

### PR DESCRIPTION
**Pull request by etungsten**
**Issue number:**
N/A

**Description of changes:**

**Testing done:**
Can build the tools image locally on my development machine.
I can use the updated versions of eksctl and eksctl-anywhere to create clusters locally.
```
    tools: update eks-anywhere to v0.12.2
    
    This updates eks-anywhere to version 0.12.2 to enable EKS 1.24 cluster
    creation.

```
```
    tools: update eksctl to v0.120
    
    This updates `eksctl` that's bundled with the testsys tools image to
    version v0.120 which allows us to manage EKS 1.24 clusters.

```

The eks-resource agent works as expected when it needs to create a cluster:
```
$ kubectl logs -f test-eks-resoura287f95b-4fd1-470d-893e-b3563d64d163-creatid88jx -n testsys
[2022-11-28T17:55:56Z INFO  resource_agent::agent] Initializing Agent
[2022-11-28T17:55:56Z INFO  eks_resource_agent::eks_provider] Beginning creation of EKS cluster 'test-eks-resource-agent' with creation policy 'Some(IfNotExists)'
[2022-11-28T17:55:56Z INFO  eks_resource_agent::eks_provider] Creation policy is 'IfNotExists' and cluster 'test-eks-resource-agent' does not exist: creating cluster
[2022-11-28T17:55:56Z INFO  eks_resource_agent::eks_provider] Creating cluster with eksctl
2022-11-28 17:55:56 [ℹ]  eksctl version 0.120.0
2022-11-28 17:55:56 [ℹ]  using region us-west-2
2022-11-28 17:55:56 [ℹ]  setting availability zones to [us-west-2b us-west-2c us-west-2a]
2022-11-28 17:55:56 [ℹ]  subnets for us-west-2b - public:192.168.0.0/19 private:192.168.96.0/19
2022-11-28 17:55:56 [ℹ]  subnets for us-west-2c - public:192.168.32.0/19 private:192.168.128.0/19
2022-11-28 17:55:56 [ℹ]  subnets for us-west-2a - public:192.168.64.0/19 private:192.168.160.0/19
2022-11-28 17:55:56 [ℹ]  nodegroup "ng-1763fb6f" will use "ami-02ff2f92f13947cbf" [AmazonLinux2/1.21]
2022-11-28 17:55:56 [ℹ]  using Kubernetes version 1.21
2022-11-28 17:55:56 [ℹ]  creating EKS cluster "test-eks-resource-agent" in "us-west-2" region with un-managed nodes
2022-11-28 17:55:56 [ℹ]  will create 2 separate CloudFormation stacks for cluster itself and the initial nodegroup
2022-11-28 17:55:56 [ℹ]  if you encounter any issues, check CloudFormation console or try 'eksctl utils describe-stacks --region=us-west-2 --cluster=test-eks-resource-agent'
2022-11-28 17:55:56 [ℹ]  Kubernetes API endpoint access will use default of {publicAccess=true, privateAccess=false} for cluster "test-eks-resource-agent" in "us-west-2"
2022-11-28 17:55:56 [ℹ]  CloudWatch logging will not be enabled for cluster "test-eks-resource-agent" in "us-west-2"
2022-11-28 17:55:56 [ℹ]  you can enable it with 'eksctl utils update-cluster-logging --enable-types={SPECIFY-YOUR-LOG-TYPES-HERE (e.g. all)} --region=us-west-2 --cluster=test-eks-resource-agent'
2022-11-28 17:55:56 [ℹ]  
2 sequential tasks: { create cluster control plane "test-eks-resource-agent", 
    2 sequential sub-tasks: { 
        wait for control plane to become ready,
        create nodegroup "ng-1763fb6f",
    } 
}
2022-11-28 17:55:56 [ℹ]  building cluster stack "eksctl-test-eks-resource-agent-cluster"
2022-11-28 17:55:57 [ℹ]  deploying stack "eksctl-test-eks-resource-agent-cluster"
...
```

cluster deletion:
```
...
2022-11-28 09:58:45 [ℹ]  deleting EKS cluster "test-eks-resource-agent"
2022-11-28 09:58:46 [ℹ]  1 task: { delete cluster control plane "test-eks-resource-agent" [async] }
2022-11-28 09:58:47 [ℹ]  will delete stack "eksctl-test-eks-resource-agent-cluster"
2022-11-28 09:58:47 [✔]  all cluster resources were deleted
...
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by webern — APPROVED** [review]

**Comment by ecpullen** [comment]
Did you also rebuild the eks-resource agent and test it?

File: tools/hashes/eksctl [hunk 1]
```diff
@@ -1,5 +1,5 @@
...
-17446c1c529e40948a388b3ae75c84dabe19ef2e3c446a95a20062039e01182d313df99b1203fff4cc944ad6dccea9df0fcf6b33fe8998ae7217b7a939638dea  eksctl_anywhere_0.12.1.tar.gz
-f59fc52ab809abc0687ec3743e4880207ccbba3881fb30b7fc9e8f3380e5fc815b5439357d592d902f6dd4d8088c3094ca0c271b8762222ad51d81baa1a062dd  eksctl_anywhere_0.12.1_23_linux_amd64.tar.gz
+53031a48bf3545a44cde2b0a8dce11d30bb465c107a4e672142e53878d02caec0a18582f0e7aa01408f4a1d282a6742586d87c42856d2bb5e2d759c9495bc86d  eksctl_0.120.0.tar.gz
+170f5a37611c3bc7c99650751df8a30cfb9514d854b45aabb6b33842ec4497fddec415f80a4af78a995577cdc71742dc6cf5c6c25bab2eb0c17cbd6a254408fd  eksctl_0.120.0_linux_amd64.tar.gz
+fea2381e3ed230462cf677adc603da0c54c5186325f24c2d36c90e8f49daf0f22d976e5c9c8b7e0a775fb5e0370763b817a8d1b5bb152964e21625d3bd4bc21f  eksctl_0.120.0_linux_arm64.tar.gz
+75a007999c45c405013bea56900a5b1facd08c4d3e40b60251fd57106dd28d19ac9ae1c39c269f25d7c7d4c41e805454b5169c5da71d1baf3eca5e6f038d90a2  eksctl_anywhere_0.12.2.tar.gz
+527ebc61d18d6d46e511219c215f4746a03cfe2fffa2447ed1344beec0baffcc7f2fa828bc0a414e877b1bcc13710b7de315ee6eb6c277065a3553a69d452dcc  eksctl_anywhere_0.12.2_24_linux_amd64.tar.gz
```
Comment (ecpullen) [inline tools/hashes/eksctl:5]: Does eksctl anywhere not have arm binaries?

Comment (etungsten) [reply tools/hashes/eksctl:5]: No they do not provide ARM binaries. I added a note here:
https://github.com/bottlerocket-os/bottlerocket-test-system/blob/123e1d18a79c8d81f20134e35dbee8311b9768b4/tools/Dockerfile#L56-L57

**Review by ecpullen — APPROVED** [review]

**Comment by etungsten** [comment]
> rebuild the eks-resource agent and test it?

I'm rebuilding the eks-resource-agent to check.

**Comment by etungsten** [comment]
Tested the `eks-resource-agent` and it create/deletes clusters fine. Updated testing description.

**Pull request merged**